Add collections tasks

tsbonev opened the pull request.

mgenov commented on src/com/clouway/crm/core/MostCommon.java:
```diff
@@ -0,0 +1,69 @@
+
+        for (String letter: wordSet
+                ) {
+            long count = Stream.of(words.split("")).filter(c -> c.equals(letter)).count();
```
Doesn't look optimized as for each letter you are going to split all words then to filter each letter in each word. 

Better usage here is with map.

mgenov commented on src/com/clouway/crm/core/MostCommon.java:
```diff
@@ -0,0 +1,69 @@
+     * @param words the string to sreach through
+     * @return a string with the result
+     */
+    public String wordCount(String words){
```
The format of the result should not be determined by this method. It is better result to be returned as structure/object and the caller to decide how to make it. 

In this particular case the result may be returned as Map as it's pair of key/value. 

```java
public Map<String, Integer> wordCount(String word) {
   //
}

// This is going to return 
Map<String, Integer> counters = wordCounter.count("word1 word2 word1 word3 word1") 
// the result should be:
//               "word1" -> 3
//               "word2" -> 1
//               "word3" -> 1
```

In this case it's a role of the client to decide how to deal with the results.

mgenov commented on src/com/clouway/crm/core/Main.java:
```diff
@@ -0,0 +1,83 @@
+        System.out.println();
+
+        Scanner in = new Scanner (System.in);
+        while(true){
```
The commented code could be removed.

tsbonev commented:
Added maps to wordCount and letterCount classes
Removed PageBean code that was commented out
Added a println stream that prints out the set from the wordCount map that gets returned

mgenov commented on src/com/clouway/crm/core/PageBean.java:
```diff
@@ -0,0 +1,137 @@
+     */
+    public PageBean(ArrayList<T> constructList, int pageSize){
+        this.pageSize = pageSize;
+        list = (List<T>)constructList.clone();
```
clone usage is not a good idea as it is going to create a new list which is holding the references to the existing values, e.g meaning becomes misleading unless users of the code explicitly declare clone function. 

```java
List<User> users = new ArrayList<User>();
users.add(new User("John"));

List<User> copyOfUsers = (List<User>)users.clone()
copyOfUsers.get(0).setName("Test") // this line is going to modify the user, so it's not a clone !
```

It's better usage to become:
```java
this.list = new ArrayList<>(source) 
```

as you can see the cast is not required in this case.

mgenov commented on src/com/clouway/crm/core/PageBean.java:
```diff
@@ -0,0 +1,137 @@
+     * @param constructList list to clone in the instance
+     * @param pageSize the size of every page
+     */
+    public PageBean(ArrayList<T> constructList, int pageSize){
```
Always pass the upper interface and not a concrete implementation as it provides implementation detail. 

In this case it is the List interface.

tsbonev commented:
Changed the parameter to pass a List<> instead of an ArrayList<>
Changed the copying code to use a new instance of ArrayList instead of cloning the references of the old one
